app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CategoriesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_icons, only: %i[new create]
+
+  def new
+    @category = Category.new
+    @category.icon = @icons.first
```

Comment:
Hello @joel-kalema, I appreciate your suggestion, and you're right, it would be a better approach. However, the current project requirements don't mention it explicitly. I'll take your advice into consideration and plan to implement it over the weekend. Thank you for your input.